crates/biome_js_analyze/src/lint/nursery/no_magic_numbers.rs

```diff
@@ -0,0 +1,240 @@
+                    let value_token = literal.value_token().ok()?;
+                    let value_text = value_token.text();
+                    let value = value_text.parse::<i64>().ok()?;
+                    if is_allowed_number(literal.syntax(), value, config) {
+                        return None;
+                    }
+
+                    let is_const_violation = check_const_violation(literal.syntax(), config);
+
+                    Some(NumberContext {
+                        range: literal.range(),
+                        raw: value_token.text().to_string(),
+                        is_const_violation,
+                    })
+                } else {
+                    return None;
+                }
+            }
+            AnyJsExpression::JsUnaryExpression(unary) => {
+                if let Ok(operator) = unary.operator() {
+                    if operator == JsUnaryOperator::Minus {
+                        if let Ok(argument) = unary.argument() {
+                            if let Some(literal_expr) = argument.as_any_js_literal_expression() {
+                                if let Ok(value_token) = literal_expr.value_token() {
+                                    let value_text = value_token.text();
+                                    let value = value_text.parse::<i64>().ok()?;
+                                    let neg_value = -value;
+                                    if is_allowed_number(unary.syntax(), neg_value, config) {
+                                        return None;
+                                    }
+                                    let is_const_violation =
+                                        check_const_violation(unary.syntax(), config);
+                                    return Some(NumberContext {
+                                        range: unary.range(),
+                                        raw: format!("-{}", value_token.text()),
+                                        is_const_violation,
+                                    });
+                                }
+                            }
+                        }
+                    }
+                }
+                None
+            }
+            _ => None,
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let diagnostic = if state.is_const_violation {
+            RuleDiagnostic::new(
+                rule_category!(),
+                state.range,
+                markup! { "Number constants declarations must use 'const'" },
+            )
+        } else {
+            RuleDiagnostic::new(
+                rule_category!(),
+                state.range,
+                markup! { "No magic number: "{state.raw} },
+            )
+            .note(markup! {
+                "Consider extracting this magic number into a named constant."
+            })
+        };
+        Some(diagnostic)
+    }
+}
+
+fn is_allowed_number(
```

Comment:
Remove the commented code and add some documentation to the function. We want to explain the business logic of "when a number is allowed"